src/ITensorsExtensions/ITensorsExtensions.jl

```diff
@@ -57,7 +59,13 @@ function map_itensor(
 )
   USV = svd(A, lind; kwargs...)
   U, S, V, spec, u, v = USV
-  S = map_diag(s -> f(s + regularization), S)
+
+  if !isnothing(regularization)
+    S = map_diag(s -> f(s + regularization), S)
+  else
+    S = map_diag(s -> f(s), S)
+  end
+
   sqrtDL, δᵤᵥ, sqrtDR = sqrt_decomp(S, u, v)
   sqrtDR = denseblocks(sqrtDR) * denseblocks(δᵤᵥ)
   L, R = U * sqrtDL, V * sqrtDR
```

Comment:
Is all of this logic needed or can it just return `U * S * V`? Perhaps I am missing something about what this code is doing.